docs/howto_gpu.md

```diff
@@ -0,0 +1,182 @@
+# How to set up a GPU Docker container that runs
+
+While training Deep Learning algorithms, you might need GPUs to speed up the process.
+This tutorial will show you how to:
+
+- Create a Docker Image with GPU support
+- Set up Vertex AI to run the Docker Image on a GPU
+- Make sure the Vertex training component step access the GPU
+
+## Create a Docker Image with GPU support
+
+Please refer to this excellent [tutorial](https://towardsdatascience.com/a-complete-guide-to-building-a-docker-image-serving-a-machine-learning-system-in-production-d8b5b0533bde#:~:text=3.%20Building%20a%20Docker%20image%20for%20any%20Python%20Project%20(GPU)%3A) to create a Docker Image with GPU support.
+
+### 1. Identify the CUDA version of your local environment
+
+This take-away from the article is important: 
+> Always use the same CUDA and cuDNN version in Docker image as present in the underlying host machine
+
+So the first step is to identify which CUDA version and PyTorch version you are using in your local environment.
+
+To identify the CUDA version of your local environment, run:
+
+```bash
+$ pip freeze | grep cu
+
+nvidia-cudnn-cu11==8.5.0.96
+```
+
+Means I have CUDA 11 installed, and CuDNN 8.
+
+### 2. Select a Docker image that matches your CUDA version
+
+> Docker hub of Nvidia has a lot of images, so understanding their tags and selecting the correct image is the most important building block.
+
+Prefers `nvidia/cuda` Docker images compared to `pytorch/` and  `vertex-ai/training`.
+
+You need to select a Docker image that matches this version.
+
+In my case, I found this one: `nvidia/cuda:11.7.0-cudnn8-runtime-ubuntu20.04`
+
+so I would start my dockerfile with:
+
+```Dockerfile
+FROM nvidia/cuda:11.7.0-cudnn8-runtime-ubuntu20.04
+```
+
+### 3. Install Python
+
+You need to install the same Python version as the one you are using in your local environment.
+
+In my case, I am using Python 3.9, so I would add this line to my Dockerfile:
+
+```Dockerfile
+FROM nvidia/cuda:11.7.0-cudnn8-runtime-ubuntu20.04
+ENV PYTHON_VERSION=3.9
+ENV CONTAINER_TIMEZONE=Europe/Paris
+
+RUN apt update \
+    && apt install --no-install-recommends -y build-essential \
+    tzdata \
+    python${PYTHON_VERSION} \
+    python${PYTHON_VERSION}-venv \
+    python3-pip \
+    python3-setuptools \
+    python3-distutils \ 
+    && apt clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN ln -s -f /usr/bin/python${PYTHON_VERSION} /usr/bin/python3 && \
+    ln -s -f /usr/bin/python${PYTHON_VERSION} /usr/bin/python && \
+    ln -s -f /usr/bin/pip3 /usr/bin/pip
+```
```

Comment:
Formating tip: Using mkdocs you can highlight specific lines inside code blocs -> https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#highlighting-specific-lines